conda_rpms/install.py

```diff
@@ -516,18 +710,55 @@ def link(pkgs_dir, prefix, dist, linktype=LINK_HARD, index=None, target_prefix=N
         return
 
     source_dir = join(pkgs_dir, dist)
-    if not run_script(dist, 'pre-link', prefix, target_prefix):
+    if not run_script(prefix, dist, 'pre-link', target_prefix):
         sys.exit('Error: pre-link failed: %s' % dist)
 
     info_dir = join(source_dir, 'info')
     files = list(yield_lines(join(info_dir, 'files')))
     has_prefix_files = read_has_prefix(join(info_dir, 'has_prefix'))
     no_link = read_no_link(info_dir)
 
+    noarch = False
+    # If the distribution is noarch, it will contain a `link.json` file in
+    # the info_dir
+    with open(join(info_dir, 'index.json'), 'r') as fh:
+        index_data = json.loads(fh.read())
+    if 'noarch' in index_data:
+        noarch = index_data['noarch']
+
+        if on_win:
+            raise ValueError('Windows is not supported.')
+
+        link_json = join(info_dir, 'link.json')
+        if exists(link_json):
+            with open(link_json, 'r') as fh:
+                link_data = json.loads(fh.read())
+            if 'noarch' in link_data:
+                noarch_json = link_data['noarch']
+
+        target_py_version = get_python_version(prefix)
+        target_python_short_path = join('bin', 'python{}'.format(
+            target_py_version))
+        target_site_packages = join('lib', 'python{}'.format(
+            target_py_version), 'site-packages')
+
+    # A list of the files, including pyc files and entrypoints, that will be
+    # added to the metadata.
+    all_files = []
+
     with Locked(prefix), Locked(pkgs_dir):
         for f in files:
             src = join(source_dir, f)
-            dst = join(prefix, f)
+
+            if noarch == 'python':
```

Comment:
@lbdreyer If you agree with the above [comment](https://github.com/SciTools-incubator/conda-rpms/pull/9/files#r178893221) then it might be best if this was just:
```python
if noarch:
```

Reply:
but noarch='generic' requires different handling so i'd rather keep it as it is

Reply:
but noarch='generic' requires different handling so i'd rather keep it as it is

Reply:
@lbdreyer Agreed. Handling `generic` is different, so we can opt to not handle that at the moment.

However, if `noarch` is set, it can now be either set to `python` or `True` (due to legacy `noarch_python` usage in the `index.json`), so we'd want this condition to handle this otherwise we'll get odd behaviour for `noarch_python: True`